whois-api/src/main/java/net/ripe/db/whois/api/httpserver/dos/IpBlockListFilter.java

```diff
@@ -0,0 +1,65 @@
+package net.ripe.db.whois.api.httpserver.dos;
+
+import com.google.common.net.InetAddresses;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import net.ripe.db.whois.common.hazelcast.HazelcastBlockedIps;
+import net.ripe.db.whois.common.ip.IpInterval;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+
+@Component
+public class IpBlockListFilter implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IpBlockListFilter.class);
+
+    private final HazelcastBlockedIps hazelcastBlockedIps;
+
+    public IpBlockListFilter(final HazelcastBlockedIps hazelcastBlockedIps){
+        this.hazelcastBlockedIps = hazelcastBlockedIps;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain) throws IOException, ServletException {
+        final HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+        final HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
+
+        if (isBlockedIp(httpRequest.getRemoteAddr())){
+            httpResponse.setStatus(HttpStatus.TOO_MANY_REQUESTS_429);
+            httpResponse.getWriter().write("You have been permanently blocked. Please contact support");
+            return;
+        }
```

Comment:
we should use  response.sendError?

Reply:
doesn't work property with UPDATE and DELETE
the text gets disappear and replaced by "".

I created a test in WhoisRestBasicAuthTestIntegration with same issue. I think that we have a bug here in the code when using sendError()

Reply:
SendError is from jetty server, so should be doing right thing. Usually PUT and DELETE never expects a message body for erros. Can you check in case of put and delete error what we do in rest of code ? My guess is we do not send back  any message body

Reply:
shall I remove the DELETE and PUT from my ITs?

Reply:
Why remove ? You should check status code is sent properly or not if it is put or delete and ip is blocked

Reply:
I mean remove the body check from those ITs